### PR TITLE
A different approach to capturing french responses

### LINKED ISSUE #49: Create duplicate box for free text fields to enable English and French answers
So that the templates will be created in English and French, we need a way for the user to input free text answers in both languages.

### PR DESCRIPTION
**Pull request by smcmurtry**
Here's an alternative to #54 that I think is much better. 

Closes #49

### CODE REVIEW

**Comment by sastels** [comment]
:( agree about the other PR being really long. With this new approach, though, the English text is required even if they want to only do French forms :( I think we should talk to an interaction designer - basically what we are trying to do is something I haven't seen before - it's hard to know what to do.

**Comment by smcmurtry** [comment]
If someone wants to do just French forms, they could answer the questions in French and skip the translation page. So the use case you're describing is someone who isn't fluent enough to answer the questions in French but will be conducting research only in French. I doubt many researchers would fall into that category. Also, if they are not that fluent in French they would probably be writing the goals, etc. in English first and then translating, so this wouldn't be extra work.

An alternative approach could be to add another question page at the start where we ask what forms you want to generate. But that would make the process longer for everyone.

File: locales/fr.json [hunk 1]
```diff
@@ -137,5 +137,9 @@
   "No recording": "Aucun enregistrement",
   "questions-2.title": "questions-2.title",
   "questions-2.intro": "questions-2.intro",
-  "footer.link1.desc": "footer.link1.desc"
+  "footer.link1.desc": "footer.link1.desc",
+  "asdf": "asdf",
+  "form.translation": "English translation of: \"{{ text }}\"",
+  "translations-1.title": "translations-1.title",
+  "translations-1.intro": "translations-1.intro"
```
Comment (sastels) [inline locales/fr.json]: Is the French for "asdf" really "asdf"?
`form.translation` and `translations-1.*` could use some translating too.

File: routes/questions-1/questions-1.controller.js [hunk 2]
```diff
@@ -12,7 +13,7 @@ module.exports = app => {
     .get(route.path, (req, res) => {
       const jsFiles = ['js/toggle-area.js']
       res.render(name, {
-        ...routeUtils.getViewData(req, { data: req.query }),
+        ...routeUtils.getViewData(req, { data: req.query, language: `_${i18n.getLocale(req)}` }),
```
Comment (sastels) [inline routes/questions-1/questions-1.controller.js]: cool!

File: routes/questions-1/questions-1.njk [hunk 3]
```diff
@@ -27,19 +27,19 @@
                 {{ radioButtons('is_with_partner', { 'on':'Yes','off':'No'}, data.is_with_partner, 'form.is_with_partner', errors) }}
             </div>
             <div class="is_with_partner-toggled" style="display:none" >
-                {{ textInput('partner_department_name', 'form.partner_department_name') }}
-                {{ textInput('partner_department_acronym', 'form.partner_department_acronym') }}
-                {{ textInput('partner_department_authority', 'form.partner_department_authority', { hint: 'form.partner_department_authority.desc' }) }}
+                {{ textInput('partner_department_name' + language, 'form.partner_department_name') }}
```
Comment (sastels) [inline routes/questions-1/questions-1.njk]: this is a really nice way to do this.

File: routes/translations-1/translations-1.njk [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        <p>
+            {{ __('translations-1.intro') }}
+        </p>
+
+
+        <form method="post" class="form">
+
+            <input type="hidden" name="_csrf" value="{{ csrfToken }}">
+
+            {% if data["partner_department_name"+language] !== "" %}
```
Comment (sastels) [inline routes/translations-1/translations-1.njk]: We might be able to do all these lines with some sort of for loop over a list of strings? Something to poke at in a future PR if we're bored.

File: views/macros/input-text.njk [hunk 5]
```diff
@@ -7,9 +7,9 @@
     {% set firstError = errors[errors.keys()[0]] %}
 {% endif %}
 
-{% macro textInput(name, label, attributes) %}
+{% macro textInput(name, label, attributes, interpolation={}) %}
```
Comment (sastels) [inline views/macros/input-text.njk]: poking at the basic macros from `node-starter-app` makes me a bit scared for the future but there's probably not a better way to do this.

Comment (smcmurtry) [reply views/macros/input-text.njk]: Yeah there should be a supported way to do this, maybe a future PR to the starter repo.

**Pull request merged**